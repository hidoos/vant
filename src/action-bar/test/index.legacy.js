import ActionBar from '..';
import Button from '../../action-bar-button';
import Icon from '../../action-bar-icon';
import { mount } from '../../../test';

test('Button click event', () => {
  const wrapper = mount(Button);
  wrapper.trigger('click');
  expect(wrapper.emitted('click').length).toEqual(1);
});

test('Icon click event', () => {
  const wrapper = mount(Icon);
  wrapper.trigger('click');
  expect(wrapper.emitted('click').length).toEqual(1);
});

test('Button render default slot', () => {
  const wrapper = mount({
    render(h) {
      return h(Button, null, ['Default Content']);
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('Icon render default slot', () => {
  const wrapper = mount({
    render(h) {
      return h(Icon, null, ['Default Content']);
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('Icon render icon slot', () => {
  const wrapper = mount({
    render(h) {
      return h(Icon, {
        scopedSlots: {
          default: () => 'Text',
          icon: () => 'Custom Icon',
        },
      });
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('Icon render icon slot with badge', () => {
  const wrapper = mount({
    render(h) {
      return h(Icon, {
        props: {
          badge: '1',
        },
        scopedSlots: {
          default: () => 'Text',
          icon: () => 'Custom Icon',
        },
      });
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('Icon render icon slot with dot', () => {
  const wrapper = mount({
    render(h) {
      return h(Icon, {
        props: {
          dot: true,
        },
        scopedSlots: {
          default: () => 'Text',
          icon: () => 'Custom Icon',
        },
      });
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('disable safe-area-inset-bottom prop', () => {
  const wrapper = mount(ActionBar, {
    propsData: {
      safeAreaInsetBottom: false,
    },
  });

  expect(wrapper).toMatchSnapshot();
});
