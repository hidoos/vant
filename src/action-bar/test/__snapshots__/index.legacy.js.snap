// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render default slot 1`] = `
<button class="van-button van-button--default van-button--large van-action-bar-button van-action-bar-button--first van-action-bar-button--last">
  <div class="van-button__content"><span class="van-button__text">Default Content</span></div>
</button>
`;

exports[`Icon render default slot 1`] = `
<div role="button" tabindex="0" class="van-action-bar-icon">
  <div class="van-icon van-icon-undefined van-action-bar-icon__icon">
    <!---->
  </div>Default Content
</div>
`;

exports[`Icon render icon slot 1`] = `
<div role="button" tabindex="0" class="van-action-bar-icon">
  <div class="van-action-bar-icon__icon">Custom Icon
    <!---->
  </div>Text
</div>
`;

exports[`Icon render icon slot with dot 1`] = `
<div role="button" tabindex="0" class="van-action-bar-icon">
  <div class="van-action-bar-icon__icon">Custom Icon<div class="van-badge van-badge--dot"></div>
  </div>Text
</div>
`;

exports[`Icon render icon slot with badge 1`] = `
<div role="button" tabindex="0" class="van-action-bar-icon">
  <div class="van-action-bar-icon__icon">Custom Icon<div class="van-badge">1</div>
  </div>Text
</div>
`;

exports[`disable safe-area-inset-bottom prop 1`] = `<div class="van-action-bar van-action-bar--unfit"></div>`;
